dotnet/src/SemanticKernel/Configuration/KernelConfig.cs

```diff
@@ -15,9 +18,78 @@ namespace Microsoft.SemanticKernel.Configuration;
 public sealed class KernelConfig
 {
     /// <summary>
-    /// Global retry logic used for all the backends
+    /// Retry configuration for IHttpRetryPolicy that uses RetryAfter header when present.
     /// </summary>
-    public IRetryMechanism RetryMechanism { get => this._retryMechanism; }
+    public sealed class HttpRetryConfig
+    {
+        /// <summary>
+        /// Maximum number of retries.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when value is negative.</exception>
+        public int MaxRetryCount
+        {
+            get { return this._maxRetryCount; }
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(this.MaxRetryCount), "Max retry count cannot be negative.");
+                }
+
+                this._maxRetryCount = value;
+            }
+        }
+
+        /// <summary>
+        /// Minimum delay between retries.
+        /// </summary>
+        public TimeSpan MinRetryDelay { get; set; } = TimeSpan.FromSeconds(2);
+
+        /// <summary>
+        /// Maximum delay between retries.
+        /// </summary>
+        public TimeSpan MaxRetryDelay { get; set; } = TimeSpan.FromSeconds(60);
+
+        /// <summary>
+        /// Maximum total time spent retrying.
+        /// </summary>
+        public TimeSpan MaxTotalRetryTime { get; set; } = TimeSpan.FromMinutes(2);
+
+        /// <summary>
+        /// Whether to use exponential backoff or not.
+        /// </summary>
+        public bool UseExponentialBackoff { get; set; }
+
+        /// <summary>
+        /// List of status codes that should be retried.
+        /// </summary>
+        public List<HttpStatusCode> RetryableStatusCodes { get; set; } = new()
+        {
+            HttpStatusCode.RequestTimeout,
+            HttpStatusCode.ServiceUnavailable,
+            HttpStatusCode.GatewayTimeout,
+            HttpStatusCode.TooManyRequests
+        };
+
+        /// <summary>
+        /// List of exception types that should be retried.
+        /// </summary>
+        public List<Type> RetryableExceptionTypes { get; set; } = new()
+        {
+            typeof(TimeoutException),
+            typeof(WebException),
+            typeof(HttpRequestException)
```

Comment:
Can this be filtered more -- i.e. server cert invalidation

Reply:
Yeah, HttpRequestException will have inner exception like SocketException or IOException, with a subset of SocketException being things that represent DNS error or TCP timeout, etc. This could be filtered but I'd suggest we just handle all HttpRequestException and allow users to supply their own handler as appropriate. This is what Polly does.